drivers/ethernet/eth_mcux.c

```diff
@@ -166,6 +177,64 @@ rx_buffer[CONFIG_ETH_MCUX_RX_BUFFERS][ETH_MCUX_BUFFER_SIZE];
 static u8_t __aligned(ENET_BUFF_ALIGNMENT)
 tx_buffer[CONFIG_ETH_MCUX_TX_BUFFERS][ETH_MCUX_BUFFER_SIZE];
 
+#ifdef CONFIG_NET_POWER_MANAGEMENT
+static void eth_mcux_phy_enter_reset(struct eth_context *context);
+void eth_mcux_phy_stop(struct eth_context *context);
+
+static int eth_mcux_device_pm_control(struct device *dev, u32_t command,
+				      void *context, device_pm_cb cb, void *arg)
+{
+	struct eth_context *eth_ctx = (struct eth_context *)dev->driver_data;
+	int ret = 0;
+
+	if (!eth_ctx->clock_dev) {
+		LOG_ERR("No CLOCK dev");
+
+		ret = -EIO;
+		goto out;
+	}
+
+	if (command == DEVICE_PM_SET_POWER_STATE) {
+		if (*(u32_t *)context == DEVICE_PM_SUSPEND_STATE) {
+			LOG_DBG("Suspending");
+
+			ret = net_if_suspend(eth_ctx->iface);
+			if (ret == -EBUSY) {
+				goto out;
```

Comment:
I would change this test of ret to:

if (!ret) 

This will prevent the code from running twice and stop the crash when accessing the enet registers